src/components/tx/SignOrExecuteForm/ExecuteForm.tsx

```diff
@@ -159,8 +159,8 @@ const ExecuteForm = ({
           {/* Submit button */}
           <CheckWallet allowNonOwner={onlyExecute}>
             {(isOk) => (
-              <Button variant="contained" type="submit" disabled={!isOk || submitDisabled}>
-                Execute
+              <Button variant="contained" type="submit" disabled={!isOk || submitDisabled} sx={{ width: '112px' }}>
+                {submitDisabled ? <CircularProgress size={20} /> : 'Execute'}
```

Comment:
There are cases when the button is legitimately disabled, e.g. when you’re not an owner. It shouldn’t be loading in those cases.

I’d recommend checking !isSubmittable instead of submitDisabled. And it might be better to rename it to isExecuting.